android/app/src/main/java/org/fossasia/openevent/viewmodels/MainActivityViewModel.java

```diff
@@ -0,0 +1,219 @@
+package org.fossasia.openevent.viewmodels;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+import android.arch.lifecycle.Transformations;
+import android.arch.lifecycle.ViewModel;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.fossasia.openevent.OpenEventApp;
+import org.fossasia.openevent.R;
+import org.fossasia.openevent.api.APIClient;
+import org.fossasia.openevent.data.Event;
+import org.fossasia.openevent.data.Microlocation;
+import org.fossasia.openevent.data.Session;
+import org.fossasia.openevent.data.SessionType;
+import org.fossasia.openevent.data.Speaker;
+import org.fossasia.openevent.data.Sponsor;
+import org.fossasia.openevent.data.Track;
+import org.fossasia.openevent.data.extras.SocialLink;
+import org.fossasia.openevent.dbutils.LiveRealmDataObject;
+import org.fossasia.openevent.dbutils.RealmDataRepository;
+import org.fossasia.openevent.events.EventDownloadEvent;
+import org.fossasia.openevent.events.JsonReadEvent;
+import org.fossasia.openevent.events.MicrolocationDownloadEvent;
+import org.fossasia.openevent.events.RetrofitError;
+import org.fossasia.openevent.events.SessionDownloadEvent;
+import org.fossasia.openevent.events.SessionTypesDownloadEvent;
+import org.fossasia.openevent.events.SpeakerDownloadEvent;
+import org.fossasia.openevent.events.SponsorDownloadEvent;
+import org.fossasia.openevent.events.TracksDownloadEvent;
+import org.fossasia.openevent.utils.ConstantStrings;
+import org.fossasia.openevent.utils.DateConverter;
+import org.fossasia.openevent.utils.SharedPreferencesUtil;
+import org.fossasia.openevent.utils.Utils;
+
+import java.util.List;
+
+import io.reactivex.Completable;
+import io.reactivex.Observable;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.schedulers.Schedulers;
+import io.realm.Realm;
+import io.realm.RealmList;
+import io.realm.RealmResults;
+import timber.log.Timber;
+
+public class MainActivityViewModel extends ViewModel {
+    private RealmDataRepository realmRepo;
+    private LiveData<Event> eventLiveData;
+    private Event event;
+    private boolean newInstance = false;
+
+    public MainActivityViewModel() {
+        realmRepo = RealmDataRepository.getDefaultInstance();
+    }
+
+    public LiveData<Event> getEventSync() {
+        newInstance = false;
+        if (eventLiveData == null) {
+            event = realmRepo.getEventSync();
+            if (event == null) {
```

Comment:
Can event be null even after assigning it a value just on a line above it ?

Reply:
I get a NPE in the viewModel as soon as I remove the null check